core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueServiceImpl.java

```diff
@@ -175,29 +181,67 @@ public CompletableFuture<CheckStatusChangePermittedResponse> checkStatusChangePe
     }
     var issueKey = params.getIssueKey();
     var serverApi = serverApiOpt.get();
+
+
+
     return asUUID(issueKey)
       .flatMap(localOnlyIssueRepository::findByKey)
       .map(r -> {
+        // For anticipated issues we currently don't get the information from SonarQube (as there is no web API
+        // endpoint) regarding the available transitions. SonarCloud doesn't provide it currently anyway.
+        // -> Here we just rely on the SonarQube version alone. God have mercy with the poor soul who has to change it
+        //    once SonarCloud is supporting anticipated transitions or SonarQube wants to change it once again :>
         var anticipateTransitionsSupported = checkAnticipatedStatusChangeSupported(serverApi, connectionId);
-        // no way to easily check if 'Administer Issue' permission is granted, might fail later
-        return CompletableFuture.completedFuture(toResponse(anticipateTransitionsSupported, UNSUPPORTED_SQ_VERSION_REASON));
+        var is104orNewer = !serverApi.isSonarCloud() && is104orNewer(connectionId, serverApi);
+        return CompletableFuture.completedFuture(toResponse(anticipateTransitionsSupported, is104orNewer));
       })
       .orElseGet(() -> serverApi.issue().searchByKey(params.getIssueKey())
-        .thenApply(issue -> toResponse(hasAdministerIssuePermission(issue), STATUS_CHANGE_PERMISSION_MISSING_REASON)));
+        .thenApply(issue -> toResponse(getAdministerIssueTransitions(issue))));
+  }
+
+  /** For checking whether SonarQube is already on 10.4 or not. NEVER apply to SonarCloud as their version differs! */
+  private boolean is104orNewer(String connectionId, ServerApi serverApi) {
+    var serverVersionSynchronizer = new ServerInfoSynchronizer(storageService.connection(connectionId));
+    var serverVersion = serverVersionSynchronizer.readOrSynchronizeServerInfo(serverApi);
+    return serverVersion.getVersion().compareToIgnoreQualifier(SQ_ACCEPTED_TRANSITION_MIN_VERSION) >= 0;
   }
 
-  private static CheckStatusChangePermittedResponse toResponse(boolean permitted, String reason) {
-    return new CheckStatusChangePermittedResponse(permitted,
-      permitted ? null : reason,
-      // even if not permitted, return the possible statuses, if clients still want to show users what's supported
-      Arrays.asList(ResolutionStatus.values()));
+  /** Create a response for normal issues already found on SonarQube */
+  private static CheckStatusChangePermittedResponse toResponse(List<ResolutionStatus> statuses) {
+    var permitted = !statuses.isEmpty();
+
+    // When it is not permitted, we won't return the possible statuses as we don't know which statuses exist: SonarCloud
+    // might use the new transitions, might not - we don't know. For SonarQube, we could determine the available
+    // transitions but in order to have the "same" behaviour for SonarQube / SonarCloud we just return none.
```

Comment:
If SLCORE does not return possible statuses when the user does not have the correct permissions, then we can completely remove the explicit version check.

* On future SonarCloud and SonarQube 10.4+, the permission is granted if both `accept` and `falsepositive` are there; in this case, the allowed transitions are `Accept` and `False positive`
* On current SonarCloud and pre-10.4 SonarQube, the permission is granted if both `wontfix` and `falsepositive` are there; in this case, the allowed transitions are `Won't fix` and `False positive`

So IMHO, the code should:
* Check whether `accept` is in the available transitions
* If it is there, also check for `falsepositive`, and eventually return `Accept` and `False positive`
* If _not_, check for `wontfix` and `falsepositive`

This would work for all current and foreseeable future server products.